src/io/flutter/module/settings/FlutterCreateParams.java

```diff
@@ -22,4 +33,16 @@ public boolean isOfflineSelected() {
     return createProjectOfflineCheckBox.isSelected();
   }
 
+  private static boolean isPubAvailable() {
+    // Check to see if the pub site is accessible as an indicator for whether we're online
+    // and if we should expect pub commands to succeed.
+    try {
+      new Socket(InetAddress.getByName("pub.dartlang.org"), 80);
```

Comment:
Do we need to close this socket?
Probably not very important anyway, we don't create projects often.

Reply:
Ah.  But why not. Will fix.